### PR TITLE
wayland: implement modifiers

### PR DESCRIPTION
**Pull request by vberger**

### CODE REVIEW

File: src/api/wayland/keyboard.rs [hunk 1]
```diff
@@ -37,8 +38,19 @@ impl wayland_kbd::Handler for KbdHandler {
...
             let mut guard = eviter.lock().unwrap();
-	    // TODO implement ModifiersState
-            guard.push_back(Event::KeyboardInput(state, rawkey as u8, vkcode, ModifiersState::default()));
+            guard.push_back(
+                Event::KeyboardInput(
+                    state,
+                    rawkey as u8,
+                    vkcode,
+                    ModifiersState {
+                        shift: mods.shift ^ mods.caps_lock,
```
Comment (vberger) [inline src/api/wayland/keyboard.rs]: I just want confirmation about this part.

Wayland-kbd exposes the shift and caps lock key independently, and the logical equivalent of the shift modifier is represented by this XOR. Is it the proper abstraction level?

Comment (vberger) [inline src/api/wayland/keyboard.rs]: Actually, the X11 backend does not merge shift and caps lock modifiers. This is IMO sensible, as most keymaps on linux react differently to shift and caps lock.

However, the X11 backend here simply discards the caps lock state... @rigtorp did you chose this same behaviour for all platforms?

**Comment by rigtorp** [comment]
The use case for this is to be able to implement keybindings such as
Ctrl-Shift-L etc. If you merge caps lock state into shift, then
Ctrl-Capslock-L would be equivalent to Ctrl-Shift-L which i believe is
wrong. As I implemented it ModifiersState.shift is only true when left or
right shift is held down.

**Comment by vberger** [comment]
Okay, I see. I'll mimmick that then.

**Pull request merged**